### PR TITLE
Configurable watchdog and batching

### PR DESCRIPTION
**Pull request by engenegr**
Addresses https://github.com/ACINQ/eclair/issues/2260 and https://github.com/ACINQ/eclair/issues/2259.
Not tested yet on mainnet node with remote Bitcoind.

### CODE REVIEW

File: eclair-core/src/main/resources/reference.conf [hunk 1]
```diff
@@ -347,7 +348,7 @@ eclair {
       }
       safety-checks {
         // A set of basic checks on data to make sure we use the correct database
-        // Those checks are disabled by default because they wouldn't pass on a fresh new node with
+        // Those checks are disabled by default because they wouldnt pass on a fresh new node with
```
Comment (t-bast) [inline eclair-core/src/main/resources/reference.conf:351]: The previous sentence was correct, this one isn't, revert?

File: eclair-core/src/main/resources/reference.conf [hunk 2]
```diff
@@ -29,6 +29,7 @@ eclair {
     wallet = ""
     zmqblock = "tcp://127.0.0.1:29000"
     zmqtx = "tcp://127.0.0.1:29000"
+    batching = true
```
Comment (t-bast) [inline eclair-core/src/main/resources/reference.conf:32]: Maybe rename and document:

```suggestion
    // Batching requests saves bandwidth but may slightly degrade latency and reliability.
    // You may want to disable this when bitcoin is running on a remote machine with an unreliable network.
    batch-requests = true
```

File: eclair-core/src/main/resources/reference.conf [hunk 3]
```diff
@@ -382,6 +383,7 @@ eclair {
   blockchain-watchdog {
     // all available blockchain watchdog sources
     // you can remove items from this list to disable failing or non-needed sources
+    tolerance = 7
```
Comment (t-bast) [inline eclair-core/src/main/resources/reference.conf:386]: You should probably move that below the `sources` list, now we don't understand what the documentation above applies to. You should also add a comment to explain what that field does.

I would rename it to `missing-blocks-threshold` which is more explicit (and please rename the corresponding field in the code as well).

**Review by t-bast — COMMENTED** [review]
Thanks, that looks simple enough! I'd like to rename the fields for clarity and document them better, but otherwise looks good to me, let me know if it improves the reliability of your setup.

File: eclair-core/src/main/resources/reference.conf (hunk 1 above)
Comment (engenegr) [reply eclair-core/src/main/resources/reference.conf:351]: Sorry, I found out that, if copied entirely, your `reference.conf` leads to errors. And the error happens due to this symbol.
I will revert if you strongly disagree, of course.

File: eclair-core/src/main/resources/reference.conf (hunk 3 above)
Comment (engenegr) [reply eclair-core/src/main/resources/reference.conf]: Corrected

File: eclair-core/src/main/resources/reference.conf (hunk 1 above)
Comment (t-bast) [reply eclair-core/src/main/resources/reference.conf:353]: That's very surprising, I don't reproduce that...at least if you change it change it to proper english: `wouldn't` -> `would not`

Comment (engenegr) [reply eclair-core/src/main/resources/reference.conf]: Fixed

**Review by t-bast — APPROVED** [review]
LGTM, thanks!

**Pull request merged**